app/courses/[courseId]/flashcards/[flashcardSetId]/lecturer.tsx

```diff
@@ -540,16 +541,59 @@ function EditSideModal({
   const [text, setText] = useState(side?.text ?? "");
   const [isQuestion, setIsQuestion] = useState(side?.isQuestion ?? false);
   const [isAnswer, setIsAnswer] = useState(side?.isAnswer ?? false);
+  const [labelOverride, setLabelOverride] = useState(false);
 
   const valid =
     label.trim() != "" && text.trim() != "" && (isQuestion || isAnswer);
 
+  useEffect(() => {
+    // Do not change the label once it has been manually edited, reset when label empty
+    if (labelOverride && label === "") {
+      setLabelOverride(false);
+    } else if (labelOverride) {
+      return;
+    }
+    if (!["", "Question", "Answer"].includes(label)) {
+      setLabelOverride(true);
+      return;
+    }
+
+    // Auto-populate label
+    if (isQuestion && !isAnswer && label !== "Question") {
+      setLabel("Question");
+    } else if (!isQuestion && isAnswer && label !== "Answer") {
+      setLabel("Answer");
+    } else if (isQuestion === isAnswer) {
+      setLabel("");
```

Comment:
Warum leer? Wenn beides "true" ist, könnte man doch "Question and Answer" oder "QnA" oder Q&A" setzen. Da Flashcards beides gleichzeitig sein können.

Reply:
Wenn beide true sind, kann das aus mehreren Gründen sein und braucht wahrscheinlich einen deskriptiven Titel, deshalb wird das Feld gecleared. Hier geht es ja hauptsächlich darum den Standardfall zu vereinbaren mit einer Frage und einer Antwort.